campaignion_google_analytics/campaignion_google_analytics.module

```diff
@@ -228,3 +228,21 @@ function campaignion_google_analytics_page_build(&$page) {
   drupal_add_js(drupal_get_path('module', 'campaignion_google_analytics') . '/tracking.js');
   drupal_add_js(drupal_get_path('module', 'campaignion_google_analytics') . '/thank_you_page.js');
 }
+
+/**
+ * Implements hook_campaignion_tracking_snippets().
+ */
+function campaignion_google_analytics_campaignion_tracking_snippets() {
+  $snippets = [];
+  if ($ga_id = variable_get_value('campaignion_google_analytics_id')) {
+    $snippets['ga'] = <<<SNIPPET
+(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
+(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
+m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
+})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');
+ga('create', '$ga_id', 'auto');
```

Comment:
The quotes are needed in this version (otherwise it would be interpreted as a JS variable, or worse executed as code), but it’s probably much better to JSON encode it instead than to hard-code the quotes :thinking:

Reply:
Right, the heredoc is so long I forgot it's there :see_no_evil:

Not sure JSON encoding is worth the trouble? Maybe putting it in curly braces would make the variable easier to recognize? :thinking: